File: Backend/Go/Domain/Model/BaseHero.go
package domain_model

import (
	domain_model_base "github.com/almiranda86/superhero-go/Domain/Model/BaseModel"
	"github.com/google/uuid"
)

func CreateBaseHero(name string, privateId string) BaseHero {
	uuid := uuid.New()

	baseHero := BaseHero{
		BaseProperties: domain_model_base.BaseModel{
			Name: name,
			PrivateId: domain_model_base.PrivateId{
				PrivateId: privateId,
			},
			PublicId: domain_model_base.GuidId{
				PublicId: uuid.String(),
			},
		},
	}

	return baseHero
}

type BaseHero struct {
	BaseProperties domain_model_base.BaseModel

	CreateBaseHero func(name string, privateId string)
}

File: Backend/C#/SuperHero.Domain/Model/BaseHero.cs
using SuperHero.Domain.Model.BaseModel;
using System.Text.Json.Serialization;

namespace SuperHero.Domain.Model
{
    public record BaseHero : IGuidId
    {
        [JsonPropertyName("public_id")]
        public string PublicId { get; set; }

        [JsonIgnore]
        public int PrivateId { get; set; }

        [JsonPropertyName("name")]
        public string? Name { get; set; }

        public BaseHero(int privateId, string name)
        {
            PublicId = Guid.NewGuid().ToString();
            PrivateId = privateId;
            Name = name;
        }
    }
}
